TestSyncPodsDeletesWithNoPodInfraContainer is flaky

Seeing random failures.

```
I0316 00:07:11.888567   20733 kubelet.go:1288] probe success: bar2
--- FAIL: TestSyncPodsDeletesWithNoPodInfraContainer (0.00s)
	kubelet_test.go:110: expected []string{"list", "list", "list", "list", "inspect_container", "inspect_container", "stop", "create", "start", "inspect_container", "create", "start"}, got []string{"list", "list", "list", "stop", "create", "start", "inspect_container", "create", "start", "list", "inspect_container", "inspect_container"}
W0316 00:07:11.888986   20733 kubelet.go:1014] No ref for pod '1234'
W0316 00:07:11.889004   20733 kubelet.go:1014] No ref for pod '9876'
W0316 00:07:11.889395   20733 kubelet.go:1014] No ref for pod '1234'
W0316 00:07:11.889407   20733 kubelet.go:1014] No ref for pod '9876'
```

--- a/pkg/kubelet/dockertools/fake_docker_client.go
+++ b/pkg/kubelet/dockertools/fake_docker_client.go
@@ -19,6 +19,7 @@ package dockertools
 import (
 	"fmt"
 	"reflect"
+	"sort"
 	"sync"
 	"time"
 
@@ -63,6 +64,23 @@ func (f *FakeDockerClient) AssertCalls(calls []string) (err error) {
 		err = fmt.Errorf("expected %#v, got %#v", calls, f.called)
 	}
 
+	return
+}
+
+func (f *FakeDockerClient) AssertUnorderedCalls(calls []string) (err error) {
+	f.Lock()
+	defer f.Unlock()
+
+	var actual, expected []string
+	copy(actual, calls)
+	copy(expected, f.called)
+
+	sort.StringSlice(actual).Sort()
+	sort.StringSlice(expected).Sort()
+
+	if !reflect.DeepEqual(actual, expected) {
+		err = fmt.Errorf("expected(sorted) %#v, got(sorted) %#v", expected, actual)
+	}
 	return
 }
 

--- a/pkg/kubelet/kubelet_test.go
+++ b/pkg/kubelet/kubelet_test.go
@@ -121,6 +121,13 @@ func verifyCalls(t *testing.T, fakeDocker *dockertools.FakeDockerClient, calls [
 	}
 }
 
+func verifyUnorderedCalls(t *testing.T, fakeDocker *dockertools.FakeDockerClient, calls []string) {
+	err := fakeDocker.AssertUnorderedCalls(calls)
+	if err != nil {
+		t.Error(err)
+	}
+}
+
 func verifyStringArrayEquals(t *testing.T, actual, expected []string) {
 	invalid := len(actual) != len(expected)
 	if !invalid {
@@ -819,7 +826,7 @@ func TestSyncPodsDeletesWithNoPodInfraContainer(t *testing.T) {
 	}
 	waitGroup.Wait()
 
-	verifyCalls(t, fakeDocker, []string{
+	verifyUnorderedCalls(t, fakeDocker, []string{
 		"list", "list", "list", "list", "inspect_container", "inspect_container", "list", "inspect_container", "inspect_container", "stop", "create", "start", "inspect_container", "create", "start", "list", "inspect_container", "inspect_container"})
 
 	// A map iteration is used to delete containers, so must not depend on

--- a/pkg/kubelet/kubelet_test.go
+++ b/pkg/kubelet/kubelet_test.go
@@ -27,6 +27,7 @@ import (
 	"path"
 	"reflect"
 	"regexp"
+	"sort"
 	"strconv"
 	"strings"
 	"sync"
@@ -143,23 +144,15 @@ func verifyStringArrayEquals(t *testing.T, actual, expected []string) {
 }
 
 func verifyStringArrayEqualsAnyOrder(t *testing.T, actual, expected []string) {
-	invalid := len(actual) != len(expected)
-	if !invalid {
-		for _, exp := range expected {
-			found := false
-			for _, act := range actual {
-				if exp == act {
-					found = true
-					break
-				}
-			}
-			if !found {
-				t.Errorf("Expected element %q not found in %#v", exp, actual)
-			}
-		}
-	}
-	if invalid {
-		t.Errorf("Expected: %#v, Actual: %#v", expected, actual)
+	var act, exp []string
+	copy(act, actual)
+	copy(exp, expected)
+
+	sort.StringSlice(act).Sort()
+	sort.StringSlice(exp).Sort()
+
+	if !reflect.DeepEqual(exp, act) {
+		t.Errorf("Expected(sorted): %#v, Actual(sorted): %#v", exp, act)
 	}
 }
 

